src/components/checkbox/checkbox.js

```diff
@@ -171,12 +197,48 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
+        if (indeterminateStateEnabled) {
+          setIndeterminateState();
+          // The checkbox should never be in an indeterminate="true" AND checked state.
+          // To prevent this, if the checkbox is in an indetermiante
+          // state, we skip the code below.
+          return;
+        }
+
         if(ngModelCtrl.$viewValue) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminateState() {
+        var isIndeterminate = element.attr('indeterminate') === 'true' || false;
```

Comment:
The convention we usually follow is that if the attribute is found with no value, then it is treated as `true` as this follows standard HTML attributes (readonly, disabled, etc).

So,

 - `intermediate` == true
 - `intermediate="true"` == true
 - `intermediate="false"` == false

@EladBezalel @ThomasBurleson Please correct me if I'm wrong here.